test/generator/extended_buffer_t_aottest.cpp

```diff
@@ -14,9 +14,9 @@ int main(int argc, char **argv) {
             input(x, y) = sinf(x * y + 1);
         }
     }
-    Image<float> output(10, 10, 3);
+    Image<float> output(10, 10);
 
-    fancy_buffer_t fancy_input(input);
+    fancy_buffer_t fancy_input((buffer_t *)input);
```

Comment:
I'm not sure this is worth fixing, but I find it unpleasant that this is copying the buffer_t contents, but since it is taking a non-const pointer it seems to be missing clues that is the case. One solution would be to change the fancy_buffer_t constructor to take a const ref. Another possibility would be to add a getter method (or methods) to the image class to be used in such cases to aid understandability.